examples/remarketing/add_customer_match_user_list.rb

```diff
@@ -0,0 +1,244 @@
+#!/usr/bin/env ruby
+# Encoding: utf-8
+#
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# This example uses Customer Match to create a new user list (a.k.a. audience)
+# and adds users to it.
+#
+# Note: It may take up to several hours for the list to be populated with users.
+# Email addresses must be associated with a Google account.
+# For privacy purposes, the user list size will show as zero until the list has
+# at least 1,000 users. After that, the size will be rounded to the two most
+# significant digits.
+
+require 'optparse'
+require 'google/ads/google_ads'
+require 'date'
+require 'digest'
+
+def add_customer_match_user_list(customer_id)
+  client = Google::Ads::GoogleAds::GoogleAdsClient.new
+
+  user_list = create_customer_match_user_list(client, customer_id)
+  add_users_to_customer_match_user_list(client, customer_id, user_list)
+  print_customer_match_user_list(client, customer_id, user_list)
+end
+
+def create_customer_match_user_list(client, customer_id)
+  # Creates the user list.
+  operation = client.operation.create_resource.user_list do |ul|
+    ul.name = "Customer Match List #{(Time.new.to_f * 1000).to_i}"
+    ul.description = "A list of customers that originated from email and " \
+      "physical addresses"
+    # Customer Match user lists can use a membership life span of 10000 to
+    # indicate unlimited; otherwise normal values apply.
+    # Sets the membership life span to 30 days.
+    ul.membership_life_span = 30
+    ul.crm_based_user_list = client.resource.crm_based_user_list_info do |crm|
+      crm.upload_key_type = :CONTACT_INFO
+    end
+  end
+
+  # Issues a mutate request to add the user list and prints some information.
+  response = client.service.user_list.mutate_user_lists(
+    customer_id: customer_id,
+    operations: [operation],
+  )
+
+  # Prints out some information about the newly created user list.
+  resource_name = response.results.first.resource_name
+  puts "User list with resource name #{resource_name} was created."
+
+  resource_name
+end
+
+def add_users_to_customer_match_user_list(client, customer_id, user_list)
```

Comment:
I'm not sure how we're adding users without passing in some information about the users. I think this should be refactored to take a list of emails and we can default populate it with a few fake @example.com ones.

Reply:
The example adds two users, one using email address, the other one using physical address (first name, last name, country code, postal code). It might be too many default values to add, what do you think?

Reply:
SGTM to leave this as is.